packages/vm/src/imports.rs

```diff
@@ -395,6 +395,21 @@ pub fn do_debug<A: BackendApi, S: Storage, Q: Querier>(
     Ok(())
 }
 
+/// Prints a debug message to console, including remaining gas information.
+/// This does not charge gas, so debug printing should be disabled when used in a blockchain module.
+pub fn do_debug_with_gas<A: BackendApi, S: Storage, Q: Querier>(
+    env: &Environment<A, S, Q>,
+    message_ptr: u32,
+) -> VmResult<()> {
+    if env.print_debug {
+        let message_data = read_region(&env.memory(), message_ptr, MAX_LENGTH_DEBUG)?;
+        let msg = String::from_utf8_lossy(&message_data);
+        let gas_remaining = env.get_gas_left();
+        println!("{}, gas_left: {}", msg, gas_remaining);
```

Comment:
This is something we can also easily add to the output of the current `debug` implementation (for the same reasons I mentioned in the trace PR)